test/files/run/splain.check

```diff
@@ -0,0 +1,22 @@
+warning: there was one deprecation warning (since 2.13.0); re-run with -deprecation for details
+newSource1.scala:13: error: implicit error;
+!I e: II
+ImplicitChain.g invalid because
+!I impPar3: I1
```

Comment:
I find this message a bit hard to interpret. I have the impression that Dotty provides a more helpful error message ([see an example here](https://scastie.scala-lang.org/c2BsN4fuT6ebCImGdSQBQA)). Am I right, or are there limitations in what Dotty does that are addressed by splain?

Reply:
the motivation for this notation is to be terse, concise and parsable in the face of implicit chains that are tens of levels deep. I haven't gotten any feedback on this by splain users so far, so I can't tell whether this is as useful to others as it is to me.
The test output with dummy names is certainly less readable than one where the involved names and types are recognizable, like `circe.Decoder` and `shapeless.Generic`, which is the intended use case.

One solution would certainly be to offer both verbose and terse messages, to be configured by the user.

Are you suggesting that Dotty's messages would make splain obsolete? Would that imply that it wouldn't be needed in current Scala?
I haven't looked at Dotty's implicit messages yet, does it display whole chains?

Reply:
> Are you suggesting that Dotty's messages would make splain obsolete?
> Would that imply that it wouldn't be needed in current Scala?

This is not clear yet. In any case, it’s definitely useful to work on improving error messages!

> I haven't looked at Dotty's implicit messages yet, does it display whole chains?

I have posted a comparison of messages produced by Scala and Dotty [here](https://contributors.scala-lang.org/t/better-implicit-search-errors-problematic-cases-wanted/3587/2)

Reply:
I will have to investigate some test cases myself.

In any case, implicit conversions are not a feature that is handled by splain explicitly, just FYI.

My question now is: do you want to make this PR about how error messages should look? So far the mission was to make splain available as-is – if this should be different, we should make this an explicit statement.

Reply:
> My question now is: do you want to make this PR about how error messages should look? So far the mission was to make splain available as-is – if this should be different, we should make this an explicit statement.

That’s a good point. I agree that working on how error messages should look could be done in another PR.

Reply:
marking as "resolved" because although further improvement would be welcome, it's not a blocker